Ble banner

jp-bennett opened the pull request:
Remove old BLE code, use simpler banners

jp-bennett committed:
Drop unneeded Sizeof() instances

jp-bennett committed:
Use SimpleBanner for BLE pin

jp-bennett committed:
Merge branch 'develop' into ble-Banner

thebentern committed:
Merge branch 'develop' into ble-Banner

HarukiToreda commented:
This PR fixes the issue where the Bluetooth code frame doesn't disappear when pairing is complete. Not it no longer uses a frame, it instead shows the pairing code as a notification and disappears properly when connected device is paired. Tested on Heltec V3, T114, L1, Meshtiny, and RAK4631.


![20251208_215857](https://github.com/user-attachments/assets/3b9b2780-09e4-45f7-9727-bab047c3fd61)

thebentern committed:
Merge branch 'develop' into ble-Banner

Xaositek commented:
Tested on MeshTiny and Wio Tracker L1 Pro - no issues here. @thebentern no we don't currently support larger fonts in the popups.

jp-bennett committed:
Merge branch 'develop' into ble-Banner

jp-bennett committed:
Merge branch 'develop' into ble-Banner

HarukiToreda committed:
Merge branch 'develop' into ble-Banner

HarukiToreda committed:
Support for different font sizes on notification banner

HarukiToreda commented:
I added support for different font sizes for notification banner should be good now
![20260214_232348](https://github.com/user-attachments/assets/7b3bfed5-63fe-4d51-a2a1-8201ea4940d3)

![20260214_232246](https://github.com/user-attachments/assets/3688161b-9d18-4e40-968b-e6e6dc982a8d)

HarukiToreda committed:
Merge branch 'develop' into ble-Banner

Xaositek committed:
Merge branch 'develop' into ble-Banner

Copilot committed:
Fix NRF52 BLE cppcheck shadow warning

Agent-Logs-Url: https://github.com/meshtastic/firmware/sessions/de12b52c-49d5-452a-b3fb-344724649270

Co-authored-by: thebentern <9000580+thebentern@users.noreply.github.com>

thebentern committed:
Merge branch 'develop' into ble-Banner

Xaositek committed:
Merge branch 'develop' into ble-Banner
